crates/sui-framework/src/natives/object_runtime/mod.rs

```diff
@@ -388,10 +397,19 @@ impl ObjectRuntimeState {
         mut self,
         by_value_inputs: BTreeSet<ObjectID>,
         external_transfers: BTreeSet<ObjectID>,
+        mut loaded_child_objects: BTreeMap<ObjectID, SequenceNumber>,
         child_object_effects: BTreeMap<ObjectID, ChildObjectEffect>,
+        loaded_child_objects_fixed: bool,
     ) -> Result<RuntimeResults, ExecutionError> {
         let mut wrapped_children = BTreeSet::new();
-        let mut loaded_child_objects = BTreeMap::new();
+        if !loaded_child_objects_fixed {
```

Comment:
In the past, if the child object did not have an effect (because it was created and deleted) we would not remember its loaded version. This case arises if you 

1. load a dynamic object field from storage
2. transfer it to a new dynamic object field
3. remove that new dynamic object field
4. transfer it to an address 

This would incorrectly have the object as an unwrap instead of a mutation

Reply:
Also noting for a less busy time, that I am really, *really*, **really** motivated to avoid us having to find ways to explicitly code in old bugs in the tip of our codebase.